pkg/miscellaneous/miscellaneous.go

```diff
@@ -426,25 +426,20 @@ func IsBackupBucketEmpty(snapStoreConfig *brtypes.SnapstoreConfig, logger *logru
 
 // GetInitialClusterStateIfScaleup checks if it is the Scale-up scenario and returns the cluster state either `new` or `existing`.
 func GetInitialClusterStateIfScaleup(ctx context.Context, logger logrus.Entry, clientSet client.Client, podName string, podNS string) (*string, error) {
-	//Read sts spec for updated replicas to toggle `initial-cluster-state`
-	curSts := &appsv1.StatefulSet{}
-	err := clientSet.Get(ctx, client.ObjectKey{
-		Namespace: podNS,
-		Name:      podName[:strings.LastIndex(podName, "-")],
-	}, curSts)
+	// Read etcd statefulset to check annotation or updated replicas to toggle `initial-cluster-state`
+	etcdSts, err := GetStatefulSet(ctx, clientSet, podNS, podName)
 	if err != nil {
-		logger.Warn("error fetching etcd sts ", err)
+		logger.Errorf("unable to fetch statefulset {namespace: %s, name: %s} %v", podNS, podName[:strings.LastIndex(podName, "-")], err)
 		return nil, err
 	}
 
-	if IsAnnotationPresent(curSts, ScaledToMultiNodeAnnotationKey) {
+	if IsAnnotationPresent(etcdSts, ScaledToMultiNodeAnnotationKey) {
```

Comment:
you do not really require a function for a single statement. You can keep it but it might be an overkill :)